server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -2666,4 +2666,401 @@ public void test_can_resolve_index_through_aliased_relation() throws Exception {
+        analyzed = executor.analyze("select obj['a'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("a")));
+        analyzed = executor.analyze("select obj['b'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("b")));
+        analyzed = executor.analyze("select obj['c'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("c")));
+        analyzed = executor.analyze("select obj['unknown'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("unknown")));
+
+        executor.getSessionContext().setErrorOnUnknownObjectKey(false);
+        analyzed = executor.analyze("select obj['unknown'] from (select obj from c1 union all select obj from c1) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("unknown")));
+        analyzed = executor.analyze("select obj['unknown'] from (select obj from c2 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("unknown")));
+        analyzed = executor.analyze("select obj['a'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("a")));
+        analyzed = executor.analyze("select obj['b'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("b")));
+        analyzed = executor.analyze("select obj['c'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("c")));
+        analyzed = executor.analyze("select obj['unknown'] from (select obj from c1 union all select obj from c2) alias");
+        assertThat(analyzed.outputs().size(), is(1));
+        assertThat(analyzed.outputs().get(0), isFunction("subscript", isField("obj"), isLiteral("unknown")));
+    }
```

Comment:
test cases are verbose, please let me know if this is too much. I wanted to clearly show the comparisons with the session setting set to true and false. Also to document the current bahaviours that are tricky and possibly bugs.